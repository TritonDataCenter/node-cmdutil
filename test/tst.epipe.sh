#!/bin/bash

#
# Tests that programs using exitOnEpipe() do the expected thing on EPIPE.
#

set -o pipefail
source "$(dirname ${BASH_SOURCE[0]})/common.sh"

#
# First, make sure that our test itself does the right thing.
#
node "$exdir/exitOnEpipe.js" "dontignore" | head
[[ $? -ne 0 ]] || fail "expected non-zero status when EPIPE is not handled"

#
# Now make sure our code actually does the right thing.
#
node "$exdir/exitOnEpipe.js" | head
[[ $? -eq 0 ]] || fail "expected zero status when EPIPE is handled"
