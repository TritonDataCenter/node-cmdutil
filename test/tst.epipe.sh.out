waiting for EPIPE
waiting for EPIPE
waiting for EPIPE
waiting for EPIPE
waiting for EPIPE
waiting for EPIPE
waiting for EPIPE
waiting for EPIPE
waiting for EPIPE
waiting for EPIPE
waiting for EPIPE
waiting for EPIPE
waiting for EPIPE
waiting for EPIPE
waiting for EPIPE
waiting for EPIPE
waiting for EPIPE
waiting for EPIPE
waiting for EPIPE
waiting for EPIPE
